From: Wengling Chen <feiyu2817@gmail.com>
Date: Fri, 1 Nov 2019 23:10:47 +0100
Subject: Remove dependency on com.google.android.gms.vision,
 com.google.android.gms.clearcut, com.google.android.gms.phenotype

---
  chrome/android/BUILD.gn                            |   2 -
  services/BUILD.gn                                  |   4 -
  services/shape_detection/BUILD.gn                  |   2 -
  .../shape_detection/BarcodeDetectionImpl.java      | 132 +--------------------
  .../org/chromium/shape_detection/BitmapUtils.java  |   2 -
  .../shape_detection/FaceDetectionImplGmsCore.java  | 116 ++----------------
  .../shape_detection/TextDetectionImpl.java         |  71 +----------
  third_party/android_deps/autorolled/BUILD.gn       |  56 ---------
  third_party/cardboard/BUILD.gn                     |   2 -
  9 files changed, 19 insertions(+), 368 deletions(-)

--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -313,8 +313,6 @@ if (_is_default_toolchain) {
       "$google_play_services_package:google_play_services_base_java",
       "$google_play_services_package:google_play_services_basement_java",
       "$google_play_services_package:google_play_services_tasks_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:content_uri_utils_java",
       "//base:flagged_apis_java",
       "//base:process_launcher_java",
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -207,8 +207,6 @@ if (is_android) {
       "$google_play_services_package:google_play_services_base_java",
       "$google_play_services_package:google_play_services_basement_java",
       "$google_play_services_package:google_play_services_tasks_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
@@ -250,8 +248,6 @@ if (is_android) {
     deps = [
       "$google_play_services_package:google_play_services_base_java",
       "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//mojo/public/java:base_java",
--- a/services/shape_detection/BUILD.gn
+++ b/services/shape_detection/BUILD.gn
@@ -132,8 +132,6 @@ if (is_android) {
     deps = [
       "$google_play_services_package:google_play_services_base_java",
       "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//mojo/public/java:base_java",
       "//mojo/public/java:bindings_java",
--- a/services/shape_detection/android/java/src/org/chromium/shape_detection/BarcodeDetectionImpl.java
+++ b/services/shape_detection/android/java/src/org/chromium/shape_detection/BarcodeDetectionImpl.java
@@ -4,19 +4,8 @@
 
 package org.chromium.shape_detection;
 
-import android.graphics.Point;
-import android.graphics.Rect;
-import android.util.SparseArray;
-
-import com.google.android.gms.vision.Frame;
-import com.google.android.gms.vision.barcode.Barcode;
-import com.google.android.gms.vision.barcode.BarcodeDetector;
-
-import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
 import org.chromium.build.annotations.NullMarked;
-import org.chromium.gfx.mojom.PointF;
-import org.chromium.gfx.mojom.RectF;
 import org.chromium.mojo.system.MojoException;
 import org.chromium.shape_detection.mojom.BarcodeDetection;
 import org.chromium.shape_detection.mojom.BarcodeDetectionResult;
@@ -28,101 +17,16 @@ import org.chromium.shape_detection.mojo
 public class BarcodeDetectionImpl implements BarcodeDetection {
     private static final String TAG = "BarcodeDetectionImpl";
 
-    private final BarcodeDetector mBarcodeDetector;
-
-    public BarcodeDetectionImpl(BarcodeDetectorOptions options) {
-        int formats = Barcode.ALL_FORMATS;
-        if (options.formats != null && options.formats.length > 0) {
-            formats = 0;
-            // Keep this list in sync with the constants defined in
-            // com.google.android.gms.vision.barcode.Barcode and the list of
-            // supported formats in BarcodeDetectionProviderImpl.
-            for (int i = 0; i < options.formats.length; ++i) {
-                if (options.formats[i] == BarcodeFormat.AZTEC) {
-                    formats |= Barcode.AZTEC;
-                } else if (options.formats[i] == BarcodeFormat.CODE_128) {
-                    formats |= Barcode.CODE_128;
-                } else if (options.formats[i] == BarcodeFormat.CODE_39) {
-                    formats |= Barcode.CODE_39;
-                } else if (options.formats[i] == BarcodeFormat.CODE_93) {
-                    formats |= Barcode.CODE_93;
-                } else if (options.formats[i] == BarcodeFormat.CODABAR) {
-                    formats |= Barcode.CODABAR;
-                } else if (options.formats[i] == BarcodeFormat.DATA_MATRIX) {
-                    formats |= Barcode.DATA_MATRIX;
-                } else if (options.formats[i] == BarcodeFormat.EAN_13) {
-                    formats |= Barcode.EAN_13;
-                } else if (options.formats[i] == BarcodeFormat.EAN_8) {
-                    formats |= Barcode.EAN_8;
-                } else if (options.formats[i] == BarcodeFormat.ITF) {
-                    formats |= Barcode.ITF;
-                } else if (options.formats[i] == BarcodeFormat.PDF417) {
-                    formats |= Barcode.PDF417;
-                } else if (options.formats[i] == BarcodeFormat.QR_CODE) {
-                    formats |= Barcode.QR_CODE;
-                } else if (options.formats[i] == BarcodeFormat.UPC_A) {
-                    formats |= Barcode.UPC_A;
-                } else if (options.formats[i] == BarcodeFormat.UPC_E) {
-                    formats |= Barcode.UPC_E;
-                } else {
-                    Log.e(TAG, "Unsupported barcode format hint: " + options.formats[i]);
-                }
-            }
-        }
-        mBarcodeDetector =
-                new BarcodeDetector.Builder(ContextUtils.getApplicationContext())
-                        .setBarcodeFormats(formats)
-                        .build();
-    }
+    public BarcodeDetectionImpl(BarcodeDetectorOptions options) {}
 
     @Override
     public void detect(org.chromium.skia.mojom.BitmapN32 bitmapData, Detect_Response callback) {
-        // The vision library will be downloaded the first time the API is used
-        // on the device; this happens "fast", but it might have not completed,
-        // bail in this case. Also, the API was disabled between and v.9.0 and
-        // v.9.2, see https://developers.google.com/android/guides/releases.
-        if (!mBarcodeDetector.isOperational()) {
-            Log.e(TAG, "BarcodeDetector is not operational");
-            callback.call(new BarcodeDetectionResult[0]);
-            return;
-        }
-
-        Frame frame = BitmapUtils.convertToFrame(bitmapData);
-        if (frame == null) {
-            Log.e(TAG, "Error converting Mojom Bitmap to Frame");
-            callback.call(new BarcodeDetectionResult[0]);
-            return;
-        }
-
-        final SparseArray<Barcode> barcodes = mBarcodeDetector.detect(frame);
-
-        BarcodeDetectionResult[] barcodeArray = new BarcodeDetectionResult[barcodes.size()];
-        for (int i = 0; i < barcodes.size(); i++) {
-            barcodeArray[i] = new BarcodeDetectionResult();
-            final Barcode barcode = barcodes.valueAt(i);
-            barcodeArray[i].rawValue = barcode.rawValue;
-            final Rect rect = barcode.getBoundingBox();
-            barcodeArray[i].boundingBox = new RectF();
-            barcodeArray[i].boundingBox.x = rect.left;
-            barcodeArray[i].boundingBox.y = rect.top;
-            barcodeArray[i].boundingBox.width = rect.width();
-            barcodeArray[i].boundingBox.height = rect.height();
-            final Point[] corners = barcode.cornerPoints;
-            barcodeArray[i].cornerPoints = new PointF[corners.length];
-            for (int j = 0; j < corners.length; j++) {
-                barcodeArray[i].cornerPoints[j] = new PointF();
-                barcodeArray[i].cornerPoints[j].x = corners[j].x;
-                barcodeArray[i].cornerPoints[j].y = corners[j].y;
-            }
-            barcodeArray[i].format = toBarcodeFormat(barcode.format);
-        }
-        callback.call(barcodeArray);
+        Log.e(TAG, "BarcodeDetector is not operational");
+        callback.call(new BarcodeDetectionResult[0]);
     }
 
     @Override
-    public void close() {
-        mBarcodeDetector.release();
-    }
+    public void close() {}
 
     @Override
     public void onConnectionError(MojoException e) {
@@ -130,34 +34,6 @@ public class BarcodeDetectionImpl implem
     }
 
     private int toBarcodeFormat(int format) {
-        switch (format) {
-            case Barcode.CODE_128:
-                return BarcodeFormat.CODE_128;
-            case Barcode.CODE_39:
-                return BarcodeFormat.CODE_39;
-            case Barcode.CODE_93:
-                return BarcodeFormat.CODE_93;
-            case Barcode.CODABAR:
-                return BarcodeFormat.CODABAR;
-            case Barcode.DATA_MATRIX:
-                return BarcodeFormat.DATA_MATRIX;
-            case Barcode.EAN_13:
-                return BarcodeFormat.EAN_13;
-            case Barcode.EAN_8:
-                return BarcodeFormat.EAN_8;
-            case Barcode.ITF:
-                return BarcodeFormat.ITF;
-            case Barcode.QR_CODE:
-                return BarcodeFormat.QR_CODE;
-            case Barcode.UPC_A:
-                return BarcodeFormat.UPC_A;
-            case Barcode.UPC_E:
-                return BarcodeFormat.UPC_E;
-            case Barcode.PDF417:
-                return BarcodeFormat.PDF417;
-            case Barcode.AZTEC:
-                return BarcodeFormat.AZTEC;
-        }
         return BarcodeFormat.UNKNOWN;
     }
 }
--- a/services/shape_detection/android/java/src/org/chromium/shape_detection/FaceDetectionImplGmsCore.java
+++ b/services/shape_detection/android/java/src/org/chromium/shape_detection/FaceDetectionImplGmsCore.java
@@ -5,25 +5,11 @@
 package org.chromium.shape_detection;
 import org.chromium.build.annotations.NullMarked;
 
-import android.graphics.PointF;
-import android.util.SparseArray;
-
-import com.google.android.gms.vision.Frame;
-import com.google.android.gms.vision.face.Face;
-import com.google.android.gms.vision.face.FaceDetector;
-import com.google.android.gms.vision.face.Landmark;
-
-import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
-import org.chromium.gfx.mojom.RectF;
 import org.chromium.mojo.system.MojoException;
 import org.chromium.shape_detection.mojom.FaceDetection;
 import org.chromium.shape_detection.mojom.FaceDetectionResult;
 import org.chromium.shape_detection.mojom.FaceDetectorOptions;
-import org.chromium.shape_detection.mojom.LandmarkType;
-
-import java.util.ArrayList;
-import java.util.List;
 
 /**
  * Google Play services implementation of the FaceDetection service defined in
@@ -35,110 +21,26 @@ public class FaceDetectionImplGmsCore im
     private static final int MAX_FACES = 32;
     private final int mMaxFaces;
     private final boolean mFastMode;
-    private final FaceDetector mFaceDetector;
 
     FaceDetectionImplGmsCore(FaceDetectorOptions options) {
-        FaceDetector.Builder builder =
-                new FaceDetector.Builder(ContextUtils.getApplicationContext());
         mMaxFaces = Math.min(options.maxDetectedFaces, MAX_FACES);
         mFastMode = options.fastMode;
-
-        try {
-            builder.setMode(mFastMode ? FaceDetector.FAST_MODE : FaceDetector.ACCURATE_MODE);
-            builder.setLandmarkType(FaceDetector.ALL_LANDMARKS);
-            if (mMaxFaces == 1) {
-                builder.setProminentFaceOnly(true);
-            }
-        } catch (IllegalArgumentException e) {
-            Log.e(TAG, "Unexpected exception " + e);
-            assert false;
-        }
-
-        mFaceDetector = builder.build();
     }
 
     @Override
     public void detect(org.chromium.skia.mojom.BitmapN32 bitmapData, Detect_Response callback) {
-        // The vision library will be downloaded the first time the API is used
-        // on the device; this happens "fast", but it might have not completed,
-        // bail in this case.
-        if (!mFaceDetector.isOperational()) {
-            Log.e(TAG, "FaceDetector is not operational");
-
-            // Fallback to Android's FaceDetectionImpl.
-            FaceDetectorOptions options = new FaceDetectorOptions();
-            options.fastMode = mFastMode;
-            options.maxDetectedFaces = mMaxFaces;
-            FaceDetectionImpl detector = new FaceDetectionImpl(options);
-            detector.detect(bitmapData, callback);
-            return;
-        }
-
-        Frame frame = BitmapUtils.convertToFrame(bitmapData);
-        if (frame == null) {
-            Log.e(TAG, "Error converting Mojom Bitmap to Frame");
-            callback.call(new FaceDetectionResult[0]);
-            return;
-        }
-
-        final SparseArray<Face> faces = mFaceDetector.detect(frame);
-
-        FaceDetectionResult[] faceArray = new FaceDetectionResult[faces.size()];
-        for (int i = 0; i < faces.size(); i++) {
-            faceArray[i] = new FaceDetectionResult();
-            final Face face = faces.valueAt(i);
-
-            final PointF corner = face.getPosition();
-            faceArray[i].boundingBox = new RectF();
-            faceArray[i].boundingBox.x = corner.x;
-            faceArray[i].boundingBox.y = corner.y;
-            faceArray[i].boundingBox.width = face.getWidth();
-            faceArray[i].boundingBox.height = face.getHeight();
-
-            final List<Landmark> landmarks = face.getLandmarks();
-            ArrayList<org.chromium.shape_detection.mojom.Landmark> mojoLandmarks =
-                    new ArrayList<org.chromium.shape_detection.mojom.Landmark>(landmarks.size());
-
-            for (int j = 0; j < landmarks.size(); j++) {
-                final Landmark landmark = landmarks.get(j);
-                final int landmarkType = landmark.getType();
-                if (landmarkType != Landmark.LEFT_EYE
-                        && landmarkType != Landmark.RIGHT_EYE
-                        && landmarkType != Landmark.BOTTOM_MOUTH
-                        && landmarkType != Landmark.NOSE_BASE) {
-                    continue;
-                }
-
-                org.chromium.shape_detection.mojom.Landmark mojoLandmark =
-                        new org.chromium.shape_detection.mojom.Landmark();
-                mojoLandmark.locations = new org.chromium.gfx.mojom.PointF[1];
-                mojoLandmark.locations[0] = new org.chromium.gfx.mojom.PointF();
-                mojoLandmark.locations[0].x = landmark.getPosition().x;
-                mojoLandmark.locations[0].y = landmark.getPosition().y;
-
-                if (landmarkType == Landmark.LEFT_EYE) {
-                    mojoLandmark.type = LandmarkType.EYE;
-                } else if (landmarkType == Landmark.RIGHT_EYE) {
-                    mojoLandmark.type = LandmarkType.EYE;
-                } else if (landmarkType == Landmark.BOTTOM_MOUTH) {
-                    mojoLandmark.type = LandmarkType.MOUTH;
-                } else {
-                    assert landmarkType == Landmark.NOSE_BASE;
-                    mojoLandmark.type = LandmarkType.NOSE;
-                }
-                mojoLandmarks.add(mojoLandmark);
-            }
-            faceArray[i].landmarks =
-                    mojoLandmarks.toArray(
-                            new org.chromium.shape_detection.mojom.Landmark[mojoLandmarks.size()]);
-        }
-        callback.call(faceArray);
+        Log.e(TAG, "FaceDetector is not operational");
+
+        // Fallback to Android's FaceDetectionImpl.
+        FaceDetectorOptions options = new FaceDetectorOptions();
+        options.fastMode = mFastMode;
+        options.maxDetectedFaces = mMaxFaces;
+        FaceDetectionImpl detector = new FaceDetectionImpl(options);
+        detector.detect(bitmapData, callback);
     }
 
     @Override
-    public void close() {
-        mFaceDetector.release();
-    }
+    public void close() {}
 
     @Override
     public void onConnectionError(MojoException e) {
--- a/services/shape_detection/android/java/src/org/chromium/shape_detection/TextDetectionImpl.java
+++ b/services/shape_detection/android/java/src/org/chromium/shape_detection/TextDetectionImpl.java
@@ -4,21 +4,9 @@
 
 package org.chromium.shape_detection;
 
-import android.graphics.Point;
-import android.graphics.Rect;
-import android.util.SparseArray;
-
-import com.google.android.gms.vision.Frame;
-import com.google.android.gms.vision.text.TextBlock;
-import com.google.android.gms.vision.text.TextRecognizer;
-
-import org.chromium.base.ContextUtils;
 import org.chromium.base.Log;
 import org.chromium.build.annotations.NullMarked;
 import org.chromium.build.annotations.Nullable;
-import org.chromium.gfx.mojom.PointF;
-import org.chromium.gfx.mojom.RectF;
-import org.chromium.gms.ChromiumPlayServicesAvailability;
 import org.chromium.mojo.system.MojoException;
 import org.chromium.shape_detection.mojom.TextDetection;
 import org.chromium.shape_detection.mojom.TextDetectionResult;
@@ -28,59 +16,16 @@ import org.chromium.shape_detection.mojo
 public class TextDetectionImpl implements TextDetection {
     private static final String TAG = "TextDetectionImpl";
 
-    private final TextRecognizer mTextRecognizer;
-
-    public TextDetectionImpl() {
-        mTextRecognizer = new TextRecognizer.Builder(ContextUtils.getApplicationContext()).build();
-    }
+    public TextDetectionImpl() {}
 
     @Override
     public void detect(org.chromium.skia.mojom.BitmapN32 bitmapData, Detect_Response callback) {
-        // The vision library will be downloaded the first time the API is used
-        // on the device; this happens "fast", but it might have not completed,
-        // bail in this case. Also, the API was disabled between and v.9.0 and
-        // v.9.2, see https://developers.google.com/android/guides/releases.
-        if (!mTextRecognizer.isOperational()) {
-            Log.e(TAG, "TextDetector is not operational");
-            callback.call(new TextDetectionResult[0]);
-            return;
-        }
-
-        Frame frame = BitmapUtils.convertToFrame(bitmapData);
-        if (frame == null) {
-            Log.e(TAG, "Error converting Mojom Bitmap to Frame");
-            callback.call(new TextDetectionResult[0]);
-            return;
-        }
-
-        final SparseArray<TextBlock> textBlocks = mTextRecognizer.detect(frame);
-
-        TextDetectionResult[] detectedTextArray = new TextDetectionResult[textBlocks.size()];
-        for (int i = 0; i < textBlocks.size(); i++) {
-            detectedTextArray[i] = new TextDetectionResult();
-            final TextBlock textBlock = textBlocks.valueAt(i);
-            detectedTextArray[i].rawValue = textBlock.getValue();
-            final Rect rect = textBlock.getBoundingBox();
-            detectedTextArray[i].boundingBox = new RectF();
-            detectedTextArray[i].boundingBox.x = rect.left;
-            detectedTextArray[i].boundingBox.y = rect.top;
-            detectedTextArray[i].boundingBox.width = rect.width();
-            detectedTextArray[i].boundingBox.height = rect.height();
-            final Point[] corners = textBlock.getCornerPoints();
-            detectedTextArray[i].cornerPoints = new PointF[corners.length];
-            for (int j = 0; j < corners.length; j++) {
-                detectedTextArray[i].cornerPoints[j] = new PointF();
-                detectedTextArray[i].cornerPoints[j].x = corners[j].x;
-                detectedTextArray[i].cornerPoints[j].y = corners[j].y;
-            }
-        }
-        callback.call(detectedTextArray);
+        Log.e(TAG, "TextDetector is not operational");
+        callback.call(new TextDetectionResult[0]);
     }
 
     @Override
-    public void close() {
-        mTextRecognizer.release();
-    }
+    public void close() {}
 
     @Override
     public void onConnectionError(MojoException e) {
@@ -88,11 +33,7 @@ public class TextDetectionImpl implement
     }
 
     public static @Nullable TextDetection create() {
-        if (!ChromiumPlayServicesAvailability.isGooglePlayServicesAvailable(
-                ContextUtils.getApplicationContext())) {
-            Log.e(TAG, "Google Play Services not available");
-            return null;
-        }
-        return new TextDetectionImpl();
+        Log.e(TAG, "Google Play Services not available");
+        return null;
     }
 }
--- a/third_party/android_deps/autorolled/BUILD.gn
+++ b/third_party/android_deps/autorolled/BUILD.gn
@@ -161,38 +161,6 @@ template("define_autorolled_targets") {
 
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_vision_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_vision/play-services-vision.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_vision/com_google_android_gms_play_services_vision.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_vision_common_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_vision_common_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_vision_common/play-services-vision-common.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_vision_common/com_google_android_gms_play_services_vision_common.info"
-        enable_bytecode_checks = false
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_clearcut_java",
-          "$google_play_services_package:google_play_services_flags_java",
-          "$google_play_services_package:google_play_services_phenotype_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
     if (!defined(material_design_target)) {
       android_aar_prebuilt("com_google_android_material_material_java") {
         aar_path = "autorolled/cipd/libs/com_google_android_material_material/material.aar"
@@ -792,30 +760,6 @@ template("define_autorolled_targets") {
     }
 
     # This is generated, do not edit. Update BuildConfigGenerator.groovy
-    # instead.
-    if (google_play_services_package == "//third_party/android_deps") {
-      android_aar_prebuilt("google_play_services_clearcut_java") {
-        aar_path = "autorolled/cipd/libs/com_google_android_gms_play_services_clearcut/play-services-clearcut.aar"
-        info_path = "autorolled/committed/libs/com_google_android_gms_play_services_clearcut/com_google_android_gms_play_services_clearcut.info"
-        enable_bytecode_checks = false
-
-        # To remove visibility constraint, add this dependency to
-        # //third_party/android_deps/autorolled/build.gradle.template
-        visibility = [
-          ":*",
-          "//third_party/androidx:*",
-        ]
-        deps = [
-          "$google_play_services_package:google_play_services_base_java",
-          "$google_play_services_package:google_play_services_basement_java",
-          "$google_play_services_package:google_play_services_phenotype_java",
-          "$google_play_services_package:google_play_services_tasks_java",
-          "//third_party/androidx:androidx_core_core_java",
-        ]
-      }
-    }
-
-    # This is generated, do not edit. Update BuildConfigGenerator.groovy
     # instead.
     if (google_play_services_package == "//third_party/android_deps") {
       android_aar_prebuilt("google_play_services_cloud_messaging_java") {
--- a/third_party/cardboard/BUILD.gn
+++ b/third_party/cardboard/BUILD.gn
@@ -53,8 +53,6 @@ android_library("cardboard_java") {
     ":cardboard_resources",
     "$google_play_services_package:google_play_services_base_java",
     "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_vision_common_java",
-    "$google_play_services_package:google_play_services_vision_java",
     "//third_party/android_deps:protobuf_lite_runtime_java",
     "//third_party/androidx:androidx_annotation_annotation_jvm_java",
     "//third_party/androidx:androidx_appcompat_appcompat_java",
